File: apps/api/AssetManagement.Api/Models/SystemSetting.cs
namespace AssetManagement.Api.Models;

public class SystemSetting
{
    public required string Key { get; set; }
    public required string Value { get; set; }
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public string? UpdatedBy { get; set; }
}

File: apps/api-kt/src/main/kotlin/com/assetmanagement/api/model/SystemSetting.kt
package com.assetmanagement.api.model

import jakarta.persistence.*
import java.time.Instant

@Entity
@Table(name = "system_settings")
class SystemSetting(
    @Id
    @Column(name = "`key`", nullable = false)
    var key: String = "",

    @Column(name = "value", nullable = false, columnDefinition = "TEXT")
    var value: String = "",

    @Column(name = "updated_at", nullable = false)
    var updatedAt: Instant = Instant.now(),

    @Column(name = "updated_by")
    var updatedBy: String? = null
)
